src/bin/read_csv/mod_sav.c

```diff
@@ -125,14 +125,39 @@ void produce_missingness_sav(void *csv_metadata, const char* column) {
 
 void produce_column_header_sav(void *csv_metadata, const char *column, readstat_variable_t* var) {
     struct csv_metadata *c = (struct csv_metadata *)csv_metadata;
-    metadata_column_type_t coltype = column_type(c->json_md, column, c->output_format);
-    if (coltype == METADATA_COLUMN_TYPE_DATE) {
-        var->type = READSTAT_TYPE_DOUBLE;
-        snprintf(var->format, sizeof(var->format), "%s", "EDATE40");
-    } else if (coltype == METADATA_COLUMN_TYPE_NUMERIC) {
-        var->type = READSTAT_TYPE_DOUBLE;
-        snprintf(var->format, sizeof(var->format), "F8.%d", get_decimals(c->json_md, column));
-    } else if (coltype == METADATA_COLUMN_TYPE_STRING) {
+    extract_metadata_type_t coltype = column_type(c->json_md, column, c->output_format);
+    if (coltype == EXTRACT_METADATA_TYPE_NUMERIC) {
+        extract_metadata_format_t colformat = column_format(c->json_md, column);
+        switch (colformat) {
+        case EXTRACT_METADATA_FORMAT_NUMBER:
+            var->type = READSTAT_TYPE_DOUBLE;
+            snprintf(var->format, sizeof(var->format), "F8.%d", get_decimals(c->json_md, column));
+        break;
+        case EXTRACT_METADATA_FORMAT_PERCENT:
+            var->type = READSTAT_TYPE_DOUBLE;
+            snprintf(var->format, sizeof(var->format), "F8.%d", get_decimals(c->json_md, column));
+        break;
+        case EXTRACT_METADATA_FORMAT_CURRENCY:
+            var->type = READSTAT_TYPE_DOUBLE;
+            snprintf(var->format, sizeof(var->format), "F8.%d", get_decimals(c->json_md, column));
+        break;
+        case EXTRACT_METADATA_FORMAT_DATE:
+            var->type = READSTAT_TYPE_DOUBLE;
+            snprintf(var->format, sizeof(var->format), "%s", "EDATE40");
```

Comment:
@evanmiller Same as .dta, should we include other formats for date time and time?